cfme/common/provider.py

```diff
@@ -514,12 +442,22 @@ def get_credentials(cls, credential_dict, cred_type=None):
         """
         domain = credential_dict.get('domain', None)
         token = credential_dict.get('token', None)
-        return cls.Credential(
-            principal=credential_dict['username'],
-            secret=credential_dict['password'],
-            cred_type=cred_type,
-            domain=domain,
-            token=token)
+        if not cred_type:
```

Comment:
**OPTIONAL** this looks regular, perhaps credential classes could grow a class-method to make this lookup more uniform, it may even be sensible to use the actual types for lookup instead of constants

however that change would be out of scope for the pr itself